beamer/agent.py

```diff
@@ -145,6 +145,9 @@ def _setup_direction(
             claim_request_extension=claim_request_extension,
             l1_resolutions={},
             l1_invalidations={},
+            logger=structlog.get_logger("EventProcessor").bind(
+                source_chain_id=source_chain.id, target_chain_id=target_chain.id
+            ),
```

Comment:
Few things here:

- nitpick: this would look nicer if we were to only have `logger=logger` here and assign to `logger` above the context creation line
- I don't think we should use `EventProcessor` here because this logger is going to be owned by the context; so maybe `Context`? It's not ideal either, but it's a result of the whole messy split of the event processor funcs....